wpilibcExamples/src/main/cpp/examples/StateSpaceDifferentialDriveSimulation/include/subsystems/DriveSubsystem.h

```diff
@@ -0,0 +1,191 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2019-2020 FIRST. All Rights Reserved.                        */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#pragma once
+
+#include <frc/AnalogGyro.h>
+#include <frc/Encoder.h>
+#include <frc/PWMVictorSPX.h>
+#include <frc/SpeedControllerGroup.h>
+#include <frc/drive/DifferentialDrive.h>
+#include <frc/geometry/Pose2d.h>
+#include <frc/kinematics/DifferentialDriveOdometry.h>
+#include <frc/simulation/AnalogGyroSim.h>
+#include <frc/simulation/DifferentialDrivetrainSim.h>
+#include <frc/simulation/EncoderSim.h>
+#include <frc/simulation/Field2d.h>
+#include <frc2/command/SubsystemBase.h>
+#include <units/voltage.h>
+
+#include "Constants.h"
+
+using namespace DriveConstants;
+
+class DriveSubsystem : public frc2::SubsystemBase {
+ public:
+  DriveSubsystem()
+      : m_left1{kLeftMotor1Port},
+        m_left2{kLeftMotor2Port},
+        m_right1{kRightMotor1Port},
+        m_right2{kRightMotor2Port},
+        m_leftEncoder{kLeftEncoderPorts[0], kLeftEncoderPorts[1]},
+        m_rightEncoder{kRightEncoderPorts[0], kRightEncoderPorts[1]},
+        m_gyro(0),
+        m_odometry{m_gyro.GetRotation2d()},
+        m_drivetrainSimulator{kDrivetrainPlant, kTrackwidth, kDrivetrainGearbox,
+                              kDrivetrainGearing,
+                              units::inch_t(kWheelDiameterInches / 2)},
+        m_leftEncoderSim(m_leftEncoder),
+        m_rightEncoderSim(m_rightEncoder),
+        m_gyroAngleSim(m_gyro) {
```

Comment:
These can be inline with the member variable declaration instead of in a ctor initializer list.